app/src/features/surveys/observations/sampling-sites/edit/components/SamplingSiteEditMapControl.tsx

```diff
@@ -0,0 +1,173 @@
+  const classes = useStyles();
+  const surveyContext = useContext(SurveyContext);
+  const urlParams: Record<string, string | number | undefined> = useParams();
+  const surveySampleSiteId = Number(urlParams['survey_sample_site_id']);
+
+  const sampleSiteData = surveyContext.sampleSiteDataLoader.data
+    ? surveyContext.sampleSiteDataLoader.data.sampleSites.find((x) => x.survey_sample_site_id === surveySampleSiteId)
+    : undefined;
+
+  const { name, mapId, formikProps } = props;
+
+  const { values, errors, setFieldValue, setFieldError } = formikProps;
+
+  const [updatedBounds, setUpdatedBounds] = useState<LatLngBoundsExpression | undefined>(undefined);
+  const [staticLayers, setStaticLayers] = useState<IStaticLayer[]>([]);
+
+  const boundaryUploadHandler = (): IUploadHandler => {
+    return async (file) => {
+      if (file?.type.includes('zip') || file?.name.includes('.zip')) {
+        await handleShapeFileUpload(file, name, formikProps);
+      } else if (file?.type.includes('gpx') || file?.name.includes('.gpx')) {
+        await handleGPXUpload(file, name, formikProps);
+      } else if (file?.type.includes('kml') || file?.name.includes('.kml')) {
+        await handleKMLUpload(file, name, formikProps);
+      }
+    };
+  };
+
+  const removeFile = () => {
+    setFieldValue(name, sampleSiteData?.geojson ? [sampleSiteData?.geojson] : []);
+    setFieldError(name, undefined);
+  };
+
+  // Array of sampling site features
+  const samplingSiteGeoJsonFeatures: Feature[] = get(values, name);
+
+  useEffect(() => {
+    setUpdatedBounds(calculateUpdatedMapBounds(samplingSiteGeoJsonFeatures));
+
+    const staticLayers: IStaticLayer[] = [
+      {
+        layerName: 'Sampling Sites',
+        features: samplingSiteGeoJsonFeatures.map((feature: Feature) => ({ geoJSON: feature }))
+      }
+    ];
+
+    setStaticLayers(staticLayers);
+  }, [samplingSiteGeoJsonFeatures]);
+
+  return (
+    <Grid item xs={12}>
+      <Box my={3}>
+        <FileUpload
+          uploadHandler={boundaryUploadHandler()}
+          onRemove={removeFile}
+          dropZoneProps={{
+            maxNumFiles: 1,
+            multiple: false,
+            acceptedFileExtensions: '.zip'
+          }}
+          hideDropZoneOnMaxFiles={true}
+          FileUploadItemComponent={FileUploadItem}
+          FileUploadItemComponentProps={{
+            SubtextComponent: SampleSiteFileUploadItemSubtext,
+            ActionButtonComponent: SampleSiteFileUploadItemActionButton,
+            ProgressBarComponent: SampleSiteFileUploadItemProgressBar
+          }}
+        />
+      </Box>
+      <Box component="fieldset">
+        <Typography component="legend" data-testid="funding-source-list-found">
+          Site Location Preview &zwnj;
+          <Typography component="span" color="textSecondary" fontWeight="400">
+            {samplingSiteGeoJsonFeatures.length > 0 ? `(${samplingSiteGeoJsonFeatures.length} locations detected)` : ''}
```

Comment:
Same comment re: conditionally rendering this `<span>`, and using the pluralize util.